scripts/ctest/ctest_driver.py

```diff
@@ -38,7 +40,43 @@ def _regex_matching_any(words):
     return "^(" + "|".join(words) + ")$"
 
 
-def run_single_test_suite(suite, ctest_path, cmake_build_path, build_config, disable_gpu, only_gpu, generate_xml, repeat, extra_args):
+def _get_only_failed(cmake_build_path):
+    # type (str) -> str
+    """
+    Read and parse the LastTestsFailed log if possible to build a list of tests to run. A string is returned that is
+    passed to the CTest regex -R option. The string contains all of the failed test modules to run joined with '|'.
+    :param cmake_build_path: The patch to the CMake build folder as a string
+    :return: A string to pass to CTest regex option
+    """
+    failed_tests_list_lines = []
+    failed_test_modules = []
+    # Check to see if the default LastTestsFailed log exists
+    last_test_failed_log = os.path.join(cmake_build_path, 'Testing', 'Temporary', 'LastTestsFailed.log')
+    if not os.path.exists(last_test_failed_log):
+        # Read the TAG file to get the last run date
+        date_line = rp._get_ctest_tag_content(cmake_build_path)
+        # The LastTestsFailed log contains each failed module on a separate line
+        last_test_failed_log = os.path.join(cmake_build_path, 'Testing', 'Temporary', f'LastTestsFailed_{date_line}.log')
+
+    with open(last_test_failed_log, 'r') as opened_failed_log_file:
+        failed_tests_list_lines = opened_failed_log_file.readlines()
+    
+    # Test strings will be in the format of "<Run Number>:<CTest Module Name>::TEST_RUN"
+    module_regex = '^\d+:(.+)::TEST_RUN$'
+    print("Running failed tests:")
```

Comment:
[nit] This block isn't executing these tests, only building a string. The print-messages likely shouldn't be generated here.